### ISSUE TITLE
spacemacs-layouts: Suppress warning, remove duplicate keys for layouts, leave micro state 

### ISSUE THREAD

**Issue by Andre0991** [syl20bnr/spacemacs#4579]
Hi.

First, thanks for this great package.

Here are some suggestions:

1) Suppress warning
It think it would be better to have this warning in `*Messages*` instead of opening a new window.
I replaced it intentionally, after all.

![screenshot 2016-01-12 18 34 14](https://cloud.githubusercontent.com/assets/7083061/12276123/22bacce6-b95b-11e5-93c1-af552b034371.png)

2) Remove duplicate keys
So I redefined `e`. However, the builtin still shows up:
![screenshot 2016-01-12 18 39 25](https://cloud.githubusercontent.com/assets/7083061/12276230/d4c97e5a-b95b-11e5-9d2a-7a78471e8029.png)

3) Leave micro state
It's not hard to notice that my workflow  heavily uses layouts.
One thing that bothers me a bit is that after using `SPC l o` the micro state is still active. So if I type `SPC l o e` and then `gg` to go to the beginning of the buffer, it will open the layout assigned to `g` instead. 
Of course, this is just a problem because I use a lot of layouts, but I think it's sensible to leave the micro state after choosing one, because generally we change precisely to do something.
So any change to either change the default or create an option that allows changing this behaviour?

4) Open layouts automatically on startup
How can I programatically open some layouts at startup?
I would like to execute
```
spacemacs//custom-layouts-spacemacs/custom-perspective-@Org-o
```

However, this command is not created by the macro until (I guess) I use `SPC l o`.
How do I get Spacemacs to generate all those functions?

Thanks again for the great package.

**Comment by Andre0991** [comment]
Ping @CestDiego

**Comment by duerrp** [comment]
I totally agree that the layouts micro-state should be left after choosing a layout. I shot myself in the foot several time by hitting `x` when I thought I was in normal state after switching layouts... 

Moreover the micro-state keybindings are impossible to customize (short of overwriting them)... I think we need some fix for that in the mid-term.

**Comment by TheBB** [comment]
1. Agree, easy to fix.
2. Same.
3. Agree, but I'm not sure why this happens.
4. You can call `spacemacs/custom-perspective-@Org` instead, which is generated by the call to `spacemacs|define-custom-layout`. The function you are citing is the one used by the microstate, which is just a thin wrapper.

**Comment by StreakyCobra** [comment]
3. See #4225

**Comment by Andre0991** [comment]
Thanks @TheBB, I'm using the right function now.

@StreakyCobra thanks, I'll keep an eye on that PR.

**Comment by Andre0991** [comment]
@CestDiego 
Thanks for the PR.

Also, another suggestion: making the use of the predefined layouts optional.

**Comment by CestDiego** [comment]
@Andre0991 but .... sane defaults T_T ...sure I can make that option though

**Comment by Andre0991** [comment]
Ah, I was probably ambiguous, I was referring to the `ERC` and `emacs-config`, that are builtin.

Actually the problem is that I want to use `i` for the `ERC` one instead of `E`, but both show up in `SPC l o`. I think your PR solves that, anyway.

**Comment by CestDiego** [comment]
aaahhhh....no my PR doesn solve them :( I was not smart enough to handle collisions. I'll look at it next week

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
